util/make_timeline.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+from datetime import datetime
+import re
+
+def timestamp_type(arg):
+    """Defines the timestamp input format"""
+    if re.match('\d{2}:\d{2}:\d{2}\.\d{3}', arg) is None:
+        raise argparse.ArgumentTypeError
+    return arg
+
+def parse_time(timestamp):
+    """Parses a timestamp into a datetime object"""
+    return datetime.strptime(timestamp, '%Y-%m-%dT%H:%M:%S')
+
+def parse_line_time(line):
+    """Parses the line's timestamp into a datetime object"""
+    time = parse_time(line[3:22])
+    time = time.replace(microsecond=int(line[23:29]))
+    return time
+
+parser = argparse.ArgumentParser(description = "Creates a timeline from a logged encounter")
+parser.add_argument('-f', '--file', required=True, help="The path of the log file")
+parser.add_argument('-s', '--start', required=True, type=timestamp_type, help="Timestamp of the start, e.g. '12:34:56.789")
+parser.add_argument('-e', '--end', required=True, type=timestamp_type, help="Timestamp of the end, e.g. '12:34:56.789")
+
+args = parser.parse_args()
+
+timeline_position = 0
+last_ability_time = 0
+last_entry = {'time': 0, 'ability_id': ''}
+started = False
+ignore_combatant = ['Eos', 'Selene', 'Garuda-Egi', 'Titan-Egi', 'Ifrit-Egi', 'Emerald Carbuncle', 'Ruby Carbuncle', 'Rook Autoturret', 'Bishop Autoturret', 'Demi-Bahamut', 'Earthly Star', '']
+
+with open(args.file, encoding="utf8") as file:
+    for line in file:
+        # Scan the file until the start timestamp
+        if not started and line[14:26] != args.start:
+            continue
+
+        if line[14:26] == args.end:
+            break
+
+        # We're at the start of the encounter now.
+        if not started:
+            started = True
+            last_ability_time = parse_line_time(line)
+            print('0 "Start"')
+
+        # We're looking for enemy casts
+        # These lines will start with 21 or 22, and have an NPC ID (400#####)
+        # If this isn't one, skip the line
+        if not (line[0:2] == '21' or line[0:2] == '22') or not line[37:40] == '400':
+            continue
+
+        # At this point, we have a combat line for the timeline. Set up the fields we'll use
+        line_fields = line.split('|')
+        entry = {
+            'time': parse_line_time(line),
+            'combatant': line_fields[3],
+            'ability_id': line_fields[4],
+            'ability_name': line_fields[5],
+        }
+
+        # Ignore autos, probably need a better rule than this
```

Comment:
There's not a different rule, sadly.  Autos are just some random ability just like anything else.

Reply:
There might be something like filtering the most used attack, or filtering the lowest ability id, something clever like that. I'll probably just add command line arguments to filter lines by combatant or name or id, so you can run it once, see what to ignore, and then run it with the filters to get a better result.

The Byakko timeline took about an hour with this, mostly spent simply removing extraneous lines like the state of shock sequence which has like eight actual ability uses instead of four. I didn't have to touch any of the timings though, even though I double checked them, making the process a lot easier and more feasible to do for normal modes or even dungeon bosses